blockmanager.go

```diff
@@ -966,28 +1005,15 @@ func (b *blockManager) handleBlockMsg(bmsg *blockMsg) {
 			// votes in it that were hidden from the network and which
 			// validate our parent block. We should bolt these new votes
 			// into the tx tree stake of the old block template on parent.
-			svl := b.server.chainParams.StakeValidationHeight
+			svl := b.cfg.ChainParams.StakeValidationHeight
 			if b.AggressiveMining && bmsg.block.Height() >= svl {
 				b.checkBlockForHiddenVotes(bmsg.block)
 			}
 
-			// Notify stake difficulty subscribers and prune invalidated
-			// transactions.
-			best := b.chain.BestSnapshot()
-			r := b.server.rpcServer
-			if r != nil {
-				// Update registered websocket clients on the
-				// current stake difficulty.
-				r.ntfnMgr.NotifyStakeDifficulty(
```

Comment:
Please don't remove this code from here.  I suspect the thinking is that it'll end up getting sent in response to the notification from blockchain and that is likely true, however, I believe the point of duplicating the notification here is to improve latency for time critical votes.

Before changing this, a full analysis of latency would need to be done and that should be in a separate PR.  This PR should solely be focusing on decoupling the block manager from the server **without changing any semantics**.

I'll refrain from posting on the other cases, but please go through the other instances and revert the changes.